File: 02_Object_Oriented_Programming/02_Class_Relationships/02_Association/python/main.py
"""
Driver — demonstrates Association (HAS-A, both independent).
Run: python main.py
"""

from car import Car
from person import Person


def main() -> None:
    car = Car("Tesla Model 3")
    person = Person("Alice", car)

    person.go_for_drive()

    # Re-associate with a different car
    new_car = Car("BMW i4")
    person.car = new_car
    print(f"\n{person.name} switched cars.")
    person.go_for_drive()


if __name__ == "__main__":
    main()

"""
Expected Output:
Alice is going for a drive.
Driving a Tesla Model 3

Alice switched cars.
Alice is going for a drive.
Driving a BMW i4
"""

File: 02_Object_Oriented_Programming/02_Class_Relationships/02_Association/javascript/main.js
/**
 * Driver — demonstrates Association (HAS-A, both independent).
 * Run: node main.js
 */

const Car = require("./car");
const Person = require("./person");

const car = new Car("Tesla Model 3");
const person = new Person("Alice", car);

person.goForDrive();

// Re-associate with a different car
const newCar = new Car("BMW i4");
person.car = newCar;
console.log(`\n${person.name} switched cars.`);
person.goForDrive();

/*
Expected Output:
Alice is going for a drive.
Driving a Tesla Model 3

Alice switched cars.
Alice is going for a drive.
Driving a BMW i4
*/
